File: src/main/java/com/example/rpn/AlgebraicOperators.java
package com.example.rpn;


import java.util.HashMap;
import java.util.Map;
import java.util.function.BinaryOperator;


public final class AlgebraicOperators
{
  private AlgebraicOperators() {}

  public static Map<String,BinaryOperator<Double>> OPERATORS = new HashMap<>();
  static {
    OPERATORS.put("+", (op1, op2) -> op1 + op2);
    OPERATORS.put("-", (op1, op2) -> op1 - op2);
    OPERATORS.put("*", (op1, op2) -> op1 * op2);
    OPERATORS.put("/", (op1, op2) -> op1 / op2);
  }
}

File: src/main/scala/com/example/rpn/AlgebraicOperators.scala
package com.example.rpn


object AlgebraicOperators
{
  val OPERATORS = OperatorsRegistry.empty
      .withOperator("+", (op1:Double, op2:Double) => op1 + op2)
      .withOperator("-", (op1:Double, op2:Double) => op1 - op2)
      .withOperator("*", (op1:Double, op2:Double) => op1 * op2)
      .withOperator("/", (op1:Double, op2:Double) => op1 / op2)
}
